src/common/prop.c

```diff
@@ -866,3 +869,153 @@ daos_prop_entry_cmp_acl(struct daos_prop_entry *entry1,
+	val = strtok_r(NULL, ";", &end_token);
+
+	if (strcmp(name, "label") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_LABEL;
+		rc = daos_prop_entry_set_str(entry, val, strlen(val));
+	} else if (strcmp(name, "cksum") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_CSUM;
+		rc = daos_str2csumcontprop(val);
+		if (rc < 0)
+			return rc;
+		entry->dpe_val = rc;
+		rc = 0;
+	} else if (strcmp(name, "cksum_size") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_CSUM_CHUNK_SIZE;
+		entry->dpe_val = strtoull(val, NULL, 0);
+	} else if (strcmp(name, "srv_cksum") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_CSUM_SERVER_VERIFY;
+		if (strcmp(val, "on") == 0)
+			entry->dpe_val = DAOS_PROP_CO_CSUM_SV_ON;
+		else if (strcmp(val, "off") == 0)
+			entry->dpe_val = DAOS_PROP_CO_CSUM_SV_OFF;
+		else
+			rc = -DER_INVAL;
+	} else if (strcmp(name, "dedup") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_DEDUP;
+		if (strcmp(val, "off") == 0)
+			entry->dpe_val = DAOS_PROP_CO_DEDUP_OFF;
+		else if (strcmp(val, "memcmp") == 0)
+			entry->dpe_val = DAOS_PROP_CO_DEDUP_MEMCMP;
+		else if (strcmp(val, "hash") == 0)
+			entry->dpe_val = DAOS_PROP_CO_DEDUP_HASH;
+		else
+			rc = -DER_INVAL;
+	} else if (strcmp(name, "dedup_threshold") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+		entry->dpe_val = strtoull(val, NULL, 0);
+	} else if (strcmp(name, "compression") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_COMPRESS;
+		rc = daos_str2compresscontprop(val);
+		if (rc < 0)
+			return rc;
+		entry->dpe_val = rc;
+		rc = 0;
+	} else if (strcmp(name, "encryption") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_ENCRYPT;
+		rc = daos_str2encryptcontprop(val);
+		if (rc < 0)
+			return rc;
+		entry->dpe_val = rc;
+		rc = 0;
+	} else if (strcmp(name, "rf") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_REDUN_FAC;
+		if (!strcmp(val, "0"))
+			entry->dpe_val = DAOS_PROP_CO_REDUN_RF0;
+		else if (!strcmp(val, "1"))
+			entry->dpe_val = DAOS_PROP_CO_REDUN_RF1;
+		else if (!strcmp(val, "2"))
+			entry->dpe_val = DAOS_PROP_CO_REDUN_RF2;
+		else if (!strcmp(val, "3"))
+			entry->dpe_val = DAOS_PROP_CO_REDUN_RF3;
+		else if (!strcmp(val, "4"))
+			entry->dpe_val = DAOS_PROP_CO_REDUN_RF4;
+		else {
+			D_ERROR("presently supported redundancy factors (rf) are [0-4]\n");
+			rc = -DER_INVAL;
+		}
+	} else if (strcmp(name, "ec_cell") == 0) {
+		entry->dpe_type = DAOS_PROP_CO_EC_CELL_SZ;
+		entry->dpe_val = strtoull(val, NULL, 0);
+	} else if (strcmp(name, "layout_type") == 0 || strcmp(name, "layout_version") == 0 ||
+		   strcmp(name, "rf_lvl") == 0 || strcmp(name, "max_snapshot") == 0 ||
+		   strcmp(name, "alloc_oid") == 0 || strcmp(name, "owner") == 0 ||
+		   strcmp(name, "group") == 0) {
+		D_ERROR("Property %s is read only\n", name);
+		rc = -DER_INVAL;
+	} else {
+		D_ERROR("Property %s is invalid\n", name);
+		rc = -DER_INVAL;
+	}
+
+	return rc;
+}
+
+int
+daos_prop_alloc_and_set(const char *str, daos_size_t len, daos_prop_t **_prop)
+{
+	daos_prop_t	*prop = NULL;
+	char		*save_prop = NULL, *t, *local;
```

Comment:
Style nit - The coding guidelines on the wiki say each variable should be declared on its own line.

Reply:
yea, but not sure this is followed too much.
I will change though since im repushing.